contracts/lib/TxTypes.sol

```diff
@@ -92,9 +92,9 @@ library TxTypes {
     buf = abi.encodePacked(
       uint8(TxType.Deposit),
       uint32(0),
-      _tx.accountNameHash, // account name hash
       _tx.assetId, // asset id
-      _tx.amount // state amount
+      _tx.amount, // state amount
+      _tx.accountNameHash // account name hash
     );
   }
 
```

Comment:
maybe we can add a number in the error code as well? the single `N` could be hard for us to debug 

```
    require(offset == PACKED_TX_PUBDATA_BYTES, "N");
```